packages/app/pages/Inventory/inventory.tsx

```diff
@@ -75,34 +76,94 @@ class InventoryCard extends React.Component {
    * Handler to close the details modal
    */
   private closeDetails() {
-    this.setState({showModal: false})
+    this.setState({showModal: false});
   }
 }
 
+const InventoryPage = ({resources, page, theme}) => {
+  let currentTheme = themeable(theme);
+  let pageResources = resources.slice(page * 25, (page + 1) * 25);
+
+  return (
+    <section {...currentTheme(5, "sidebar_inventory", "active")} >
+      <ul {...currentTheme(8, "sidebar_list")} >
+        {pageResources.map((item, i) => (
+          <InventoryCard theme={theme} item={item} key={i}/>
+        ))}
+      </ul>
+
+    </section>
+  );
+};
+
 /**
  * The inventory page, which contains the basic page structure
  * (menu, sidebar, sub-menus, and the entire inventory list)
  * @param {any} agent The Agent who's inventory is displayed
  * @param {any} theme The page's theme that should be used to style
  */
-const Inventory: SFC<Props> = ({ agent, theme }) => {
+class Inventory extends React.Component {
+
+  state = {
+    page: 0
+  };
+
+  private agent;
+  private theme;
+
+  constructor(props) {
+    super(props);
+    this.agent = props.agent;
+    this.theme = props.theme;
+  }
+
+  updatePage = (page) => {
+    this.setState({page});
+  };
+
+  render() {
+    let currentTheme = themeable(this.theme);
+    let numPages = Math.ceil(this.agent.ownedEconomicResources.length / 25);
+    let page = this.state.page;
+
+    return (
+      <aside {...currentTheme(1, "sidebar")} >
+        <div {...currentTheme(2, "sidebar_menu")} >
+          <h4 {...currentTheme(4, "menu_title")} >
+            Inventory List <span>{this.agent.ownedEconomicResources.length}</span>
+          </h4>
+        </div>
+
+        <PageNumbers numPages={numPages} current={page} updatePage={this.updatePage} theme={this.theme} />
```

Comment:
The definition for PageNumbers should appear before it is called

Reply:
Did not know that was a thing for JS. It's been moved up.